js/app/historyStore.js

```diff
@@ -369,6 +375,32 @@ var HistoryStore = new Lang.Class({
         }
         this.set_current_item(HistoryItem.HistoryItem.new_from_object(target_item));
     },
+
+    can_share_current_item: function () {
+        let item = this.get_current_item();
+        return (item.model &&
+                item.model instanceof Eknc.ContentObjectModel &&
+                item.model.original_uri !== null &&
+                item.model.original_uri !== "");
+    },
+
+    social_share: function (network) {
```

Comment:
Not crucial, but did you have other ideas for this method name? e.g, share_* (since this is the main action)

Reply:
you mean having one API for each social network?